py-publishing.Rmd

```diff
@@ -0,0 +1,487 @@
+# Publishing {#py-publishing}
+
+## Questions {#py-publishing-questions}
+
+```{r, child="questions/py-publishing.md"}
+```
+
+## Why should I share my work on the internet?
+
+A key part of any project is communicating what you have learned.
+You've already seen how to create documents to communicate your work with R Markdown,
+and how to host whole projects on GitHub.
+This chapter is about sharing your work through a webpage on the internet.
+Some advantages of sharing your work on a webpage include:
+
+* It's easy for your visitors --- they just need to click on a link and see your work.
+  They don't need to know anything about Jupyter Notebooks, HTML, or GitHub.
+* It provides a visually friendly and customizable landing point for people interested in your project.
+  You can easily point them to the GitHub repo if they want more details.
+* It's easy for you to make updates and those updates are immediately available to any visitors ---
+  you don't have to re-send anyone any files.
+
+By the end of this chapter,
+you'll be able to take an Jupyter Notebook that lives on your computer
+and share it with the world through a link to your own webpage.
+
+## What does it take to get a webpage online?
+
+To understand what it takes to get a webpage online,
+it helps to understand roughly what happens when you point your browser at a web address.
+When you point your browser at an address,
+for example https://merely-useful.github.io/py-publishing.html,
+the following things happen:
+
+1. The URL `https://merely-useful.github.io` points to a location on another computer
+   where the relevant files for this website reside.
+   This computer is known as the website host or server,
+   since it *hosts* the webpage files on its local storage drives
+   and *serves* them to us via the internet.
+2. The browser requests the file `py-publishing.html` from the host.
+3. The browser reads the contents of `py-publishing.html` and displays it for you in the browser window.
+
+In reverse order this process also describes what you need to do to get your own website online:
+
+1. You need an HTML file that describes what people should see on your page.
+2. You need to host the HTML file on a computer on the internet.
+3. You need a way to associate a URL with the address of your host.
+
+In this chapter,
+you'll learn a process for getting your work online that leverages what you already know ---
+creating HTML files using R Markdown (#1 above) and how to host your work in a GitHub repository (#2).
```

Comment:
What do they already know?  Sounds on the Python side they haven't seen how to create HTML yet?

Reply:
Good catch! Reworded.